### PR TITLE
Go Binary for converting yaml spec to bash variables

### PR DESCRIPTION
**Pull request by mzaneri**
This PR adds the YAML to Bash Variable parser. It is not yet integrated into the kURL install script. Note that this does not handle variables that will be executed by other go binaries, i.e. the Selinux commands.

It also adds a type in the installer CRD for EKCO, and adds the preserve-config flags for Selinux, Docker, Iptables, and Firewalld into the CRD.

It also removes $IP_ALLOC_RANGE, which is a duplicated for $POD_CIDR

### CODE REVIEW

File: scripts/common/flags.sh [hunk 1]
```diff
@@ -214,6 +214,7 @@ function flags() {
                 # allow either /16 or 16 for subnet size
                 SERVICE_CIDR_RANGE=$(echo "$_value" | sed "s/\///")
                 ;;
+            # keep this
```
Comment (areed) [inline scripts/common/flags.sh:217]: What does this comment mean?

File: kurl_util/cmd/yamltobash/main.go [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+		"Rook.Version":                    "ROOK_VERSION",
+		"SelinuxConfig.Preserve":          "PRESERVE_SELINUX_CONFIG",
+		"Velero.DisableCLI":               "VELERO_DISABLE_CLI",
+		"Velero.DisableRestic":            "VELERO_DISABLE_RESTIC",
+		"Velero.LocalBucket":              "VELERO_LOCAL_BUCKET",
+		"Velero.Namespace":                "VELERO_LOCAL_BUCKET",
+		"Velero.Version":                  "VELERO_VERSION",
+		"Weave.PodCIDR":                   "POD_CIDR",
+		"Weave.IsEncryptionDisabled":      "ENCRYPT_NETWORK",
+		//TODO handle this sed
```
Comment (areed) [inline kurl_util/cmd/yamltobash/main.go:165]: Is this done?

Comment (mzaneri) [reply kurl_util/cmd/yamltobash/main.go:165]: yes

File: scripts/common/flags.sh (hunk 1 above)
Comment (mzaneri) [reply scripts/common/flags.sh:217]: this is one of the things going to be kept in flags.sh, will be removed shortly

File: kurlkinds/pkg/apis/cluster/v1beta1/installer_types.go [hunk 3]
```diff
@@ -37,21 +37,21 @@ type InstallerSpec struct {
...
 
 type Docker struct {
-	AdditionalNoProxy          string `json:"additionalNoProxy,omitempty"`
 	BypassStorageDriverWarning bool   `json:"bypassStorageDriverWarning,omitempty"`
+	DaemonConfig               string `json:"daemonConfig,omitempty"`
 	DockerRegistryIP           string `json:"dockerRegistryIP,omitempty"`
 	HardFailOnLoopback         bool   `json:"hardFailOnLoopback,omitempty"`
 	NoCEOnEE                   bool   `json:"noCEOnEE,omitempty"`
-	NoDocker                   bool   `json:"noDocker,omitempty"`
+	PreserveConfig             bool `json:"preserveConfig,omitempty"`
```
Comment (divolgin) [inline kurlkinds/pkg/apis/cluster/v1beta1/installer_types.go:53]: need go fmt

File: kurlkinds/pkg/apis/cluster/v1beta1/installer_types.go [hunk 4]
```diff
@@ -117,42 +117,54 @@ type Registry struct {
...
 	Version              string `json:"version"`
 }
 
 type SelinuxConfig struct {
-	Selinux      string     `json:"selinux,omitempty"`
-	Type         string     `json:"type,omitempty"`
-	SemanageCmds [][]string `json:"semanageCmds,omitempty"`
-	ChconCmds    [][]string `json:"chconCmds,omitempty"`
+	ChconCmds      [][]string `json:"chconCmds,omitempty"`
+	PreserveConfig bool     `json:"preserveConfig,omitempty"`
```
Comment (divolgin) [inline kurlkinds/pkg/apis/cluster/v1beta1/installer_types.go:144]: go fmt

Comment (mzaneri) [reply kurlkinds/pkg/apis/cluster/v1beta1/installer_types.go]: fixed

**Pull request merged**